src/Main.ts

```diff
@@ -1123,9 +1123,9 @@ export class Main {
     public async getChannelInfo(
         slackChannelId: string,
         teamId: string,
-    ): Promise<ConversationsInfoResponse|void> {
-        let slackClient: WebClient|undefined;
-        let teamEntry: TeamEntry | null = null;
+    ): Promise<ConversationsInfoResponse|'channel_not_allowed'|'channel_not_found'> {
```

Comment:
Hmm, if we had to do to it this way I'd rather a `{error: string}` object response, just for extensibility reasons.

Reply:
Because this is an internal function, used only once, and the bridge isn't a library, I think a string is fine.
I considered an enum, but even that seemed overkill for just two error states with no additional info.